### PR TITLE
maps: allow same key on multiple mode

### PR DESCRIPTION
**Pull request by horriblename**
A second rewrite of keymaps cuz the last one doesn't allow different action on different modes

### CODE REVIEW

File: modules/neovim/mappings/options.nix [hunk 1]
```diff
@@ -31,59 +31,72 @@
     options =
       mapConfigOptions
       // {
+        key = mkOption {
+          type = str;
+          description = ''
+            Key that triggers this keybind.
```
Comment (diniamo) [inline modules/neovim/mappings/options.nix:37]: `The key` rather, you can also make this string one line with `""`

File: modules/neovim/mappings/options.nix [hunk 2]
```diff
@@ -31,59 +31,72 @@
...
+          '';
+        };
         mode = mkOption {
           type = either str (listOf str);
           description = ''
             The short-name of the mode to set the keymapping for. Passing an empty string is the equivalent of `:map`.
 
             See `:help map-modes` for a list of modes.
           '';
+          example = ''"nvc" for normal, visual and command mode'';
```
Comment (diniamo) [inline modules/neovim/mappings/options.nix:47]: nvc isn't a real short name from looking at the docs?

File: modules/neovim/mappings/options.nix [hunk 3]
```diff
@@ -31,59 +31,72 @@
...
         };
       };
   };
 
   # legacy stuff
-  mapOption = submodule {
+  legacyMapOption = submodule {
     options = mapConfigOptions;
   };
```
Comment (diniamo) [inline modules/neovim/mappings/options.nix:55]: Redundant comment, you can also just inline this

File: modules/neovim/mappings/options.nix [hunk 4]
```diff
@@ -31,59 +31,72 @@
...
 
   mapOptions = mode:
```
Comment (diniamo) [inline modules/neovim/mappings/options.nix:57]: Maybe name _this_ legacyMapOptions?

File: modules/wrapper/rc/config.nix [hunk 5]
```diff
@@ -41,40 +41,11 @@ in {
...
-      // mapAttrs (_: legacyMap: legacyMap // {mode = namedModes.normalVisualOp;}) cfg.maps.normalVisualOp
-      // mapAttrs (_: legacyMap: legacyMap // {mode = namedModes.visualOnly;}) cfg.maps.visualOnly
-      // mapAttrs (_: legacyMap: legacyMap // {mode = namedModes.operator;}) cfg.maps.operator
-      // mapAttrs (_: legacyMap: legacyMap // {mode = namedModes.insertCommand;}) cfg.maps.insertCommand
-      // mapAttrs (_: legacyMap: legacyMap // {mode = namedModes.lang;}) cfg.maps.lang
-      // mapAttrs (_: legacyMap: legacyMap // {mode = namedModes.command;}) cfg.maps.command;
+    maps = concatLines (map toLuaKeymap cfg.keymaps);
 
-    keymaps = concatLines (map toLuaKeymap (attrsToList (filterAttrs (_: value: value != null) maps)));
+    keymaps = maps;
```
Comment (diniamo) [inline modules/wrapper/rc/config.nix:48]: Why??

File: modules/neovim/mappings/config.nix [hunk 6]
```diff
@@ -3,32 +3,75 @@
...
+                  // {
+                    inherit key;
+                    mode = legacyMapModes.${oldMode};
+                  }
+              )
+              keybinds
+          ))
+          flatten
+        ]
+      )
```
Comment (diniamo) [inline modules/neovim/mappings/config.nix:74]: I actually quite like this compared to whatever we had before

**Review by diniamo — CHANGES_REQUESTED** [review]

File: modules/neovim/mappings/options.nix (hunk 2 above)
Comment (horriblename) [reply modules/neovim/mappings/options.nix:47]: oof, I thought nvim supports this, turns out no

**Review by diniamo — APPROVED** [review]
Should be good

**Pull request merged**